ranking/models.py

```diff
@@ -1,27 +1,41 @@
+import random
 from django.db import models
 from django.conf import settings
 from django.core.urlresolvers import reverse
+from django.utils.text import slugify
 
 
 class RankedListManager(models.Manager):
-    def for_user(self, user):
-        try:
-            return user.rankedlist
-        except RankedList.DoesNotExist:
-            ranked_list = RankedList.objects.create(owner=user)
-            ranked_list.save()
-            ranked_list.name = "{}'s Slate".format(user.get_full_name())
-            ranked_list.owner = user
-            ranked_list.slug = user.username
-            ranked_list.save()
-            return ranked_list
+    def for_request(self, request):
+        user = request.user
+
+        if user.is_authenticated:
+            try:
+                return user.rankedlist
+            except RankedList.DoesNotExist:
+                return RankedList.objects.create(
+                    name="{}'s Slate".format(user.get_full_name()),
+                    slug=user.username,
+                    owner=user)
+        else:
+            try:
+                return RankedList.objects.get(pk=request.session['ranked_list_id'])
+            except (RankedList.DoesNotExist, KeyError):
+                name = "Anon {}".format(random.randint(1000000, 2000000))
```

Comment:
uh, I guess this is really unlikely to collide, but seems a little dangerous to leave it like this forever